customization-base/src/test/java/com/azure/autorest/customization/ClassCustomizationTests.java

```diff
@@ -0,0 +1,101 @@
+package com.azure.autorest.customization;
+
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URISyntaxException;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.regex.Pattern;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ClassCustomizationTests {
+    private static final Pattern WHITESPACE = Pattern.compile("\\s*");
+    private static final Logger LOGGER = LoggerFactory.getLogger(AnnotationTests.class);
+
+    @Test
+    public void testStaticCodeBlock() throws URISyntaxException {
+
+        final String fileName = "src/main/java/foo/Foo.java";
+        final String fileContent = String.join(System.lineSeparator(),
+                "public class Foo {",
+                "    ",
+                "}");
+        final String pathToLanguageServerPlugin = Paths.get(ClassCustomizationTests.class.getResource("").toURI())
+                .resolve("../../../../../../../postprocessor")
+                .normalize()
+                .toString();
+
+        final String expectedFileContent = String.join(System.lineSeparator(),
+                "public class Foo {",
+                "    static { String a = \"foo\"; }",
+                "}");
+
+        Customization customization = new Customization() {
+            @Override
+            public void customize(LibraryCustomization libraryCustomization, Logger logger) {
+                ClassCustomization classCustomization = libraryCustomization.getPackage("foo")
+                        .getClass("Foo");
+
+                classCustomization.addStaticBlock("static { String a = \"foo\"; }");
```

Comment:
Do we need to have the static block include `static` and `{}`? Could we handle that in the API?

Reply:
Yeah, we could handle it in the API itself. I'll update the API.